lib/ansible/vault.rb

```diff
@@ -101,7 +104,13 @@ def read
 
     private
 
-    def validate_password(password, options)
+    def validate_password(password, password_file, options)
+      if !password.nil? && !password_file.nil?
+        raise Error, "Password and password_file cannot both be defined."
+      elsif !password_file.nil?
+        password = IO.binread(password_file).strip
```

Comment:
When supplying a password file to Ansible it will execute it if the executable bit is set, if we're going to support password files we need to have full parity.

Reply:
Yeah, that's the right track but you'll need to capture stdout to use as the password.